packages/@divvi/mobile/src/send/saga.ts

```diff
@@ -97,11 +97,50 @@ export function* sendPaymentSaga({
       [createStandbyTransaction]
     )
 
-    const receipt = yield* call(
-      [publicClient[networkIdToNetwork[tokenInfo.networkId]], 'waitForTransactionReceipt'],
-      { hash }
-    )
-    Logger.debug(`${TAG}/sendPaymentSaga`, 'Got send transaction receipt', receipt)
+    // After sending the transaction, navigate to the initial tab if not fromExternal
+    // This ensures that the user can't submit the same transaction again accidentally
+    // But since it is internal, they will still see the error banner if the transaction fails
+    if (!fromExternal) {
+      navigateInitialTab()
+    }
+
+    // Retry logic for fetching receipt as nodes may not be immediately updated
+    let receipt
+    let attempt = 0
+    const MAX_ATTEMPTS = 5
+    while (attempt < MAX_ATTEMPTS) {
+      try {
+        Logger.debug(`${TAG}/sendPaymentSaga`, `Fetching receipt attempt ${attempt + 1}`, hash)
+        receipt = yield* call(
+          [publicClient[networkIdToNetwork[tokenInfo.networkId]], 'waitForTransactionReceipt'],
+          { hash }
+        )
+        Logger.debug(`${TAG}/sendPaymentSaga`, 'Got send transaction receipt', receipt)
+        break // Exit on success
+      } catch (error) {
+        attempt += 1
+        if (attempt >= MAX_ATTEMPTS) {
+          Logger.error(
+            `${TAG}/sendPaymentSaga`,
+            `Failed to fetch receipt after ${MAX_ATTEMPTS} attempts`,
+            error
+          )
+          throw error
+        }
+        const backoff = Math.min(2 ** attempt * 1000, 10000) // Exponential backoff, capped at 10s
+        Logger.info(
+          `${TAG}/sendPaymentSaga`,
+          `Retrying receipt fetch in ${backoff}ms (attempt ${attempt}/${MAX_ATTEMPTS})`
+        )
+        yield* delay(backoff)
+      }
```

Comment:
Just noting this would be useful in its own `retryWithBackoff` (similar to the builtin [retry](https://redux-saga.js.org/docs/api#retrymaxtries-delay-fn-args) helper).
This way we could easily apply the same logic for swaps where we'd also benefit from this logic.